k3s-io/k3s
# Add tombstone file to etcd and catch errc etcd channel

#### Proposed Changes ####

- changes to etcd embedded in etcd to catch any error sent to the errc channel in etcd and checking for the member removed error
- write a tombstone file *~with the error code "10" when the err member removed is detected~*
- ~upon registering etcd if error code 10 is detected in tombstone file~ iif the tombstone file is detected t will backup and remove the data dir for etcd
- dont exit when etcd member is removed from the cluster
 
#### Types of Changes ####

fixing bug

#### Verification ####

- start 3 k3s server nodes
- remove one of the nodes using kubectl delete
you should be able to see the node removed however k3s doesnt exit with an error

- restart k3s on the deleted node
you should be able to see k3s node starts correctly also the etcd data dir should be backuped to etcd-"%date"

#### Linked Issues ####

https://github.com/rancher/k3s/issues/2527

#### Further Comments ####

DONT MERGE!!

depends on https://github.com/k3s-io/etcd/pull/2/files
